src/dos/cdrom_image.cpp

```diff
@@ -279,116 +282,144 @@ bool CDROM_Interface_Image::GetUPC(unsigned char& attr, char* upc)
 {
-	bool rcode = false;
+	// Guard 1: bail if the game requests playback status before actually playing
+	if (player.trackFile == nullptr) {
+#ifdef DEBUG
+		LOG_MSG("CDROM: GetAudioSub => game asked for playback position "
+		        "before playing audio (ignoring)");
+#endif
+		return false;
+	}
 
-	// Convert our running tally of track-frames played to Redbook-frames played.
-	// We round up because if our track-frame tally lands in the middle of a (fractional)
+	// Convert our running tally of played *track* frames to *Redbook* frames.
+	// We 'ceil' because if our track-frame tally lands in the middle of a (fractional)
 	// Redbook frame, then that Redbook frame would have to be considered played to produce
 	// even the smallest amount of track-frames played.  This also accurately represents
 	// the very end of a sequence where the last Redbook frame might only contain a couple
 	// PCM samples - but the entire last 2352-byte Redbook frame is needed to cover those samples.
-	const Bit32u playedRedbookFrames = static_cast<Bit32u>(ceil(
-	    static_cast<float>(REDBOOK_FRAMES_PER_SECOND * player.playedTrackFrames) / player.trackFile->getRate() ));
+	const Bit32u playedRedbookFrames = static_cast<Bit32u>(
+	    ceil(static_cast<float>(REDBOOK_FRAMES_PER_SECOND * player.playedTrackFrames)
+	    / player.trackFile->getRate()) );
 
-	// Add that to the track's starting Redbook frame to determine our absolute current Redbook frame
-	const Bit32u currentRedbookFrame = player.startRedbookFrame + playedRedbookFrames;
-
-	const char cur_track = GetTrack(currentRedbookFrame);
-	if (cur_track >= 1) {
-		track = static_cast<unsigned char>(cur_track);
-		attr = tracks[track - 1].attr;
-		index = 1;
-		frames_to_msf(currentRedbookFrame + 150, &absPos.min, &absPos.sec, &absPos.fr);
-		frames_to_msf(currentRedbookFrame - tracks[track - 1].start, &relPos.min, &relPos.sec, &relPos.fr);
-		rcode = true;
+	// Add that to the track's starting sector to determine our absolute current sector
+	const Bit32u currentSector = player.startSector + playedRedbookFrames;
+	track_const_iter track(GetTrack(currentSector));
 
+	// Guard 2: bail if the track is invalid
+	if (track == tracks.end()) {
 #ifdef DEBUG
-		LOG_MSG("%s CDROM: GetAudioSub attr=%u, track=%u, index=%u",
-		        get_time(),
-		        attr,
-		        track,
-		        index);
-		LOG_MSG("%s CDROM: GetAudioSub absoute  offset (%d), MSF=%d:%d:%d",
-		        get_time(),
-		        currentRedbookFrame + 150,
-		        absPos.min,
-		        absPos.sec,
-		        absPos.fr);
-		LOG_MSG("%s CDROM: GetAudioSub relative offset (%d), MSF=%d:%d:%d",
-		        get_time(),
-		        currentRedbookFrame - tracks[track - 1].start + 150,
-		        relPos.min,
-		        relPos.sec,
-		        relPos.fr);
+		LOG_MSG("CDROM: GetAudioSub => playback position lands outside of our tracks");
 #endif
+		return false;
 	}
-	return rcode;
+
+	track_num = track->number;
+	attr = track->attr;
+	index = 1;
```

Comment:
This is weird… GitHub colorized `index` as if it was a keyword or a function…

(seems not to be an issue for compiler, but it's curious)